src/raftstore/store/peer.rs

```diff
@@ -364,6 +364,13 @@ impl Peer {
     }
 
     pub fn check_peers(&mut self) {
+        if !self.is_leader() {
+            if !self.peer_heartbeats.is_empty() {
```

Comment:
can this reduce performance seriously? if not, I prefer keeping code simple.
@ngaut